src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
 
 #define ERROR_CODES(_) \
 	_(ER_UNKNOWN, 0,			"Unknown error") \
-	_(ER_ILLEGAL_PARAMS, 1,			"Illegal parameters, %s") \
-	_(ER_MEMORY_ISSUE, 2,			"Failed to allocate %u bytes in %s for %s") \
-	_(ER_TUPLE_FOUND, 3,			"Duplicate key exists in unique index \"%s\" in space \"%s\" with old tuple - %s and new tuple - %s") \
-	_(ER_TUPLE_NOT_FOUND, 4,		"Tuple doesn't exist in index '%s' in space '%s'") \
+	_(ER_ILLEGAL_PARAMS, 1,			"Illegal parameters", "details", STRING) \
+	_(ER_MEMORY_ISSUE, 2,			"Failed to allocate memory") \
+	_(ER_TUPLE_FOUND, 3,			"Duplicate key exists", "index", STRING, "space", STRING, "old_tuple", TUPLE, "new_tuple", TUPLE) \
+	/* ER_TUPLE_NOT_FOUND, 4, Unused */ \
 	_(ER_UNSUPPORTED, 5,			"%s does not support %s") \
-	_(ER_NONMASTER, 6,			"Can't modify data on a replication slave. My master is: %s") \
+	/* ER_NONMASTER, 6, Unused */ \
 	_(ER_READONLY, 7,			"Can't modify data on a read-only instance") \
-	_(ER_INJECTION, 8,			"Error injection '%s'") \
-	_(ER_CREATE_SPACE, 9,			"Failed to create space '%s': %s") \
-	_(ER_SPACE_EXISTS, 10,			"Space '%s' already exists") \
-	_(ER_DROP_SPACE, 11,			"Can't drop space '%s': %s") \
-	_(ER_ALTER_SPACE, 12,			"Can't modify space '%s': %s") \
-	_(ER_INDEX_TYPE, 13,			"Unsupported index type supplied for index '%s' in space '%s'") \
-	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index '%s' in space '%s': %s") \
-	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space, space '%s'") \
-	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached: %u") \
-	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key in space '%s' while secondary keys exist") \
-	_(ER_KEY_PART_TYPE, 18,			"Supplied key type of part %u does not match index part type: expected %s") \
-	_(ER_EXACT_MATCH, 19,			"Invalid key part count in an exact match (expected %u, got %u)") \
-	_(ER_INVALID_MSGPACK, 20,		"Invalid MsgPack - %s") \
-	_(ER_PROC_RET, 21,			"msgpack.encode: can not encode Lua type '%s'") \
+	_(ER_INJECTION, 8,			"Error injection", "injection_place", STRING) \
+	_(ER_CREATE_SPACE, 9,			"Failed to create space", "space", STRING, "reason", STRING) \
```

Comment:
Let's use "details" instead of "reason" everywhere.

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/errcode.h b/src/box/errcode.h
index e95bd004d..4c659897a 100644
--- a/src/box/errcode.h
+++ b/src/box/errcode.h
@@ -123,12 +123,12 @@ struct errcode_record {
 	/* ER_NONMASTER, 6, Unused */ \
 	_(ER_READONLY, 7,			"Can't modify data on a read-only instance") \
 	_(ER_INJECTION, 8,			"Error injection", "injection_place", STRING) \
-	_(ER_CREATE_SPACE, 9,			"Failed to create space", "space", STRING, "reason", STRING) \
+	_(ER_CREATE_SPACE, 9,			"Failed to create space", "space", STRING, "details", STRING) \
 	_(ER_SPACE_EXISTS, 10,			"Space already exists", "space", STRING) \
-	_(ER_DROP_SPACE, 11,			"Can't drop space", "space", STRING, "reason", STRING) \
-	_(ER_ALTER_SPACE, 12,			"Can't modify space", "space", STRING, "reason", STRING) \
+	_(ER_DROP_SPACE, 11,			"Can't drop space", "space", STRING, "details", STRING) \
+	_(ER_ALTER_SPACE, 12,			"Can't modify space", "space", STRING, "details", STRING) \
 	_(ER_INDEX_TYPE, 13,			"Unsupported index type", "index", STRING, "space", STRING) \
-	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index", "index", STRING, "space", STRING, "reason", STRING) \
+	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index", "index", STRING, "space", STRING, "details", STRING) \
 	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space", "space", STRING) \
 	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached", "limit", UINT) \
 	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key while secondary keys exist", "space", STRING) \
@@ -158,13 +158,13 @@ struct errcode_record {
 	_(ER_MORE_THAN_ONE_TUPLE, 41,		"Get() doesn't support partial keys and non-unique indexes") \
 	_(ER_ACCESS_DENIED, 42,			"Access denied") \
 	_(ER_CREATE_USER, 43,			"Failed to create user", "user", STRING, "details", STRING) \
-	_(ER_DROP_USER, 44,			"Failed to drop user", "user", STRING, "reason", STRING) \
+	_(ER_DROP_USER, 44,			"Failed to drop user", "user", STRING, "details", STRING) \
 	_(ER_NO_SUCH_USER, 45,			"User not found", "user", STRING) \
 	_(ER_USER_EXISTS, 46,                   "User already exists", "user", STRING) \
 	_(ER_CREDS_MISMATCH, 47,		"User not found or supplied credentials are invalid") \
 	_(ER_UNKNOWN_REQUEST_TYPE, 48,		"Unknown request type", "request_type", UINT) \
 	_(ER_UNKNOWN_SCHEMA_OBJECT, 49,		"Unknown object type", "object_type", STRING) \
-	_(ER_CREATE_FUNCTION, 50,		"Failed to create function", "function", STRING, "reason", STRING) \
+	_(ER_CREATE_FUNCTION, 50,		"Failed to create function", "function", STRING, "details", STRING) \
 	_(ER_NO_SUCH_FUNCTION, 51,		"Function does not exist", "function", STRING) \
 	_(ER_FUNCTION_EXISTS, 52,		"Function already exists", "function", STRING) \
 	_(ER_BEFORE_REPLACE_RET, 53,		"Invalid return value of space:before_replace trigger: expected tuple or nil") \
@@ -185,7 +185,7 @@ struct errcode_record {
 	/* ER_INVALID_ORDER, 68, Unused */ \
 	_(ER_MISSING_REQUEST_FIELD, 69,		"Missing mandatory field in request", "field", STRING) \
 	_(ER_IDENTIFIER, 70,			"Invalid identifier", "id", STRING, "details", STRING) \
-	_(ER_DROP_FUNCTION, 71,			"Can't drop function", "function_id", UINT, "reason", STRING) \
+	_(ER_DROP_FUNCTION, 71,			"Can't drop function", "function_id", UINT, "details", STRING) \
 	_(ER_ITERATOR_TYPE, 72,			"Unknown iterator type", "type", STRING) \
 	_(ER_REPLICA_MAX, 73,			"Replica count limit reached", "limit", UINT) \
 	/* ER_INVALID_XLOG, 74, Unused */ \
@@ -198,11 +198,11 @@ struct errcode_record {
 	_(ER_CROSS_ENGINE_TRANSACTION, 81,	"A multi-statement transaction can not use multiple storage engines") \
 	_(ER_NO_SUCH_ROLE, 82,			"Role is not found", "role", STRING) \
 	_(ER_ROLE_EXISTS, 83,			"Role already exists", "role", STRING) \
-	_(ER_CREATE_ROLE, 84,			"Failed to create role", "role", STRING, "reason", STRING) \
+	_(ER_CREATE_ROLE, 84,			"Failed to create role", "role", STRING, "details", STRING) \
 	_(ER_INDEX_EXISTS, 85,			"Index already exists", "index", STRING) \
 	_(ER_SESSION_CLOSED, 86,		"Session is closed") \
 	_(ER_ROLE_LOOP, 87,			"Granting role would create a loop", "role", STRING, "grantee", STRING) \
-	_(ER_GRANT, 88,				"Granting failed", "reason", STRING) \
+	_(ER_GRANT, 88,				"Granting failed", "details", STRING) \
 	_(ER_PRIV_GRANTED, 89,			"Already granted", "user", STRING, "privilege", STRING, "object_type", STRING, "object", STRING) \
 	_(ER_ROLE_GRANTED, 90,			"User already has role", "user", STRING, "role", STRING) \
 	_(ER_PRIV_NOT_GRANTED, 91,		"No such grant", "user", STRING, "privilege", STRING, "object_type", STRING, "object", STRING) \
@@ -213,7 +213,7 @@ struct errcode_record {
 	_(ER_GUEST_USER_PASSWORD, 96,		"Setting password for guest user has no effect") \
 	_(ER_TRANSACTION_CONFLICT, 97,		"Transaction has been aborted by conflict") \
 	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported privilege", "object_type", STRING, "privilege", STRING) \
-	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "function", STRING, "reason", STRING) \
+	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "function", STRING, "details", STRING) \
 	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported function language", "language", STRING, "function", STRING) \
 	_(ER_RTREE_RECT, 101,			"Coordinates number does not make a point or rectangle", "object", STRING, "dimension", UINT) \
 	_(ER_PROC_C, 102, "Error in C code") \
@@ -252,13 +252,13 @@ struct errcode_record {
 	_(ER_VY_QUOTA_TIMEOUT, 135,		"Timed out waiting for Vinyl memory quota") \
 	_(ER_PARTIAL_KEY, 136,			"Index type does not support selects via a partial key", "index_type", STRING, "index_part_count", UINT, "key_part_count", UINT) \
 	_(ER_TRUNCATE_SYSTEM_SPACE, 137,	"Can't truncate a system space", "space", STRING) \
-	_(ER_LOAD_MODULE, 138,			"Failed to dynamically load module", "module", STRING, "reason", STRING) \
+	_(ER_LOAD_MODULE, 138,			"Failed to dynamically load module", "module", STRING, "details", STRING) \
 	_(ER_VINYL_MAX_TUPLE_SIZE, 139,		"Failed to allocate tuple: tuple is too large", "tuple_size", UINT) \
 	_(ER_WRONG_DD_VERSION, 140,		"Wrong _schema version: expected 'major.minor[.patch]'") \
 	_(ER_WRONG_SPACE_FORMAT, 141,		"Wrong space format field", "fieldno", UINT, "details", STRING) \
-	_(ER_CREATE_SEQUENCE, 142,		"Failed to create sequence", "sequence", STRING, "reason", STRING) \
-	_(ER_ALTER_SEQUENCE, 143,		"Can't modify sequence", "sequence", STRING, "reason", STRING) \
-	_(ER_DROP_SEQUENCE, 144,		"Can't drop sequence", "sequence", STRING, "reason", STRING) \
+	_(ER_CREATE_SEQUENCE, 142,		"Failed to create sequence", "sequence", STRING, "details", STRING) \
+	_(ER_ALTER_SEQUENCE, 143,		"Can't modify sequence", "sequence", STRING, "details", STRING) \
+	_(ER_DROP_SEQUENCE, 144,		"Can't drop sequence", "sequence", STRING, "details", STRING) \
 	_(ER_NO_SUCH_SEQUENCE, 145,		"Sequence does not exist", "sequence", STRING) \
 	_(ER_SEQUENCE_EXISTS, 146,		"Sequence already exists", "sequence", STRING) \
 	_(ER_SEQUENCE_OVERFLOW, 147,		"Sequence overflow", "sequence", STRING) \

```

</details>